x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -167,12 +181,86 @@ func (k msgServer) ClaimMorseSupplier(
 		preClaimSupplierStake = *foundSupplier.Stake
 	}
 
+	sharedParams := k.sharedKeeper.GetParams(sdkCtx)
+	sessionEndHeight := sharedtypes.GetSessionEndHeight(&sharedParams, sdkCtx.BlockHeight())
+
+	postClaimSupplierStake := preClaimSupplierStake.Add(morseClaimableAccount.GetSupplierStake())
+	minStake := k.supplierKeeper.GetParams(ctx).MinStake
+
+	// If the claimed supplier stake is less than the minimum stake, the supplier is immediately unstaked.
+	if postClaimSupplierStake.Amount.LT(minStake.Amount) {
+		currentSessionStart := sharedtypes.GetSessionStartHeight(&sharedParams, sdkCtx.BlockHeight())
+		previousSessionEnd := sharedtypes.GetSessionEndHeight(&sharedParams, currentSessionStart-1)
+		supplier := &sharedtypes.Supplier{
+			OwnerAddress:            shannonOwnerAddr.String(),
+			OperatorAddress:         shannonOperatorAddr.String(),
+			Stake:                   &postClaimSupplierStake,
+			UnstakeSessionEndHeight: uint64(previousSessionEnd),
+			// Services:             (intentionally omitted, no services were staked),
+			// ServiceConfigHistory: (intentionally omitted, no services were staked),
+		}
+
+		// Emit an event which signals that the morse supplier has been claimed.
+		morseSupplierClaimedEvent := migrationtypes.EventMorseSupplierClaimed{
```

Comment:
Why can't we just one have one place to emit this event (for both cases)?

Reply:
This method is refactored in #1338.